src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st

```diff
@@ -60,21 +61,17 @@ OCBytecodeToASTCacheTest >> testLowerThanFirstBCOffsetAccessTest [
 { #category : #tests }
 OCBytecodeToASTCacheTest >> testNodeForBCOffsetRangeTest [
 	"As we associate each node to each possible bytecode offset that can refer to it,
-	we have to check that associations are consistent between a range and a node"
+	we have to check that associations are all pointing to a node"
 
 	| pcRange |
 	pcRange := 0 to: cache lastBcOffset.
-	pcRange do: [ :pc | 
-		self
-			assert: (cache nodeForPC: pc)
-			identicalTo: (compiledMethod sourceNodeForPC: pc) ]
+	pcRange do: [ :pc | self assert: ((cache nodeForPC: pc) isKindOf: RBProgramNode)]
 ]
 
 { #category : #tests }
 OCBytecodeToASTCacheTest >> testNodeForBCOffsetTest [
 	| pc |
+	"there should be an assignment at pc 51"
 	pc := 51.
-	self
-		assert: (cache nodeForPC: pc)
-		identicalTo: (compiledMethod sourceNodeForPC: pc)
+	self assert: (cache nodeForPC: pc) isAssignment
```

Comment:
I would add additional checks. I put that example here thinking that it refers to the assignment `i := 5` of `helper12` method (but same idea if that's the other assignment).

```Smalltalk
| pc node |
"there should be an assignment at pc 51"	
pc := 51.
node := (cache nodeForPC: pc).
self assert: node isAssignment.
self assert: node variable equals: i. "find the i var node in the method"
self assert: node variable equals: 5. "put a literal node here to check for equality"
```

Reply:
I change it to: 	

self assert: (cache nodeForPC: pc) sourceCode equals: 'i := i - 1'.